src/control/lib/ui/flags.go

```diff
@@ -89,37 +98,30 @@ func propError(fs string, args ...interface{}) *flags.Error {
 	}
 }
 
-type keyMap map[string]struct{}
-
-func (m keyMap) Keys() []string {
-	keys := make([]string, 0, len(m))
-	for key := range m {
-		keys = append(keys, key)
-	}
-	sort.Strings(keys)
-
-	return keys
-}
-
+// CompletionMap is a map of key to a list of possible completions.
 type CompletionMap map[string][]string
 
+// SetPropertiesFlag is used to hold a list of properties to set.
 type SetPropertiesFlag struct {
 	ParsedProps  map[string]string
-	settableKeys keyMap
+	settableKeys common.StringSet
 	completions  CompletionMap
 }
 
+// SettableKeys accepts a list of property keys that are settable.
 func (f *SetPropertiesFlag) SettableKeys(keys ...string) {
-	f.settableKeys = make(keyMap)
+	f.settableKeys = make(common.StringSet)
 	for _, key := range keys {
-		f.settableKeys[key] = struct{}{}
+		f.settableKeys.Add(key)
 	}
 }
 
+// SetCompletions sets the possible completions for the SetPropertiesFlag.
 func (f *SetPropertiesFlag) SetCompletions(comps CompletionMap) {
 	f.completions = comps
 }
 
+// IsSettable returns true if the key is in the list of settable keys.
 func (f *SetPropertiesFlag) IsSettable(key string) bool {
```

Comment:
Wondering if this and GetPropertiesFlag might benefit from more generic names at this point (SetKeyValueFlag, GetKeyFlag?)